model.py

```diff
@@ -5362,39 +5525,33 @@ def consolidate_works(cls, _db, calculate_work_even_if_no_author=False,
 
 
     def calculate_work(self, even_if_no_author=False, known_edition=None):
-        """Try to find an existing Work for this LicensePool.
+        """Find or create a Work for this LicensePool.
 
-        If there are no Works for the permanent work ID associated
-        with this LicensePool's presentation edition, create a new Work.
-
-        Pools that are not open-access will always have a new Work
-        created for them.
+        A pool that is not open-access will always have its own
+        Work. Open-access LicensePools will be grouped together with
+        other open-access LicensePools based on the permanent work ID
+        of the LicensePool's presentation edition.
 
         :param even_if_no_author: Ordinarily this method will refuse
         to create a Work for a LicensePool whose Edition has no title
         or author. But sometimes a book just has no known author. If
         that's really the case, pass in even_if_no_author=True and the
         Work will be created.
         """
-        self.set_presentation_edition(None)
-
-        presentation_edition = known_edition or self.presentation_edition
```

Comment:
Specifically, there's frequently a problem right here. If we always call set_presentation_edition(), there's a good chance that `self.presentation_edition` will be set to a different edition than `known_edition`. This causes the check `presentation_edition.is_presentation_for != self` to fail, because only one Edition can be the presentation edition for a LicensePool. Even if that magic Edition used to be `known_edition`, it might not be `known_edition` by the time we finish calling `set_presentation_edition().`

So in the future I want to get rid of `known_edition` and always rely on `LicensePool.presentation_edition`. But for now, I'm skipping the `set_presentation_edition` call if `known_edition` is provided.